src/current/v23.1/connection-pooling.md

```diff
@@ -25,37 +25,149 @@ Each time an application reads or writes data, it will request one of the connec
 
 Connection pooling can be a enabled as a feature of the driver, a separate library used in conjunction with a driver, a feature of an application server, or a proxy server that acts as a gateway to the database server.
 
+Creating the appropriate size pool of connections is critical to gaining maximum performance in an application. Too few connections in the pool will result in high latency as each operation waits for a connection to open up. But adding too many connections to the pool can also result in high latency as each connection thread is being run in parallel by the system. The time it takes for many threads to complete in parallel is typically higher than the time it takes a smaller number of threads to run sequentially.
+
 {{site.data.alerts.callout_success}}
 To read more about connection pooling, see our [What is Connection Pooling, and Why Should You Care](https://www.cockroachlabs.com/blog/what-is-connection-pooling/) blog post.
 {{site.data.alerts.end}}
 
-## Sizing connection pools
+## Recommended settings for connection pools
 
-Idle connections in CockroachDB do not consume many resources compared to PostgreSQL. Cockroach Labs estimates the memory overhead of idle connections in CockroachDB is 20 kB to 30 kB per connection.
+These are Cockroach Labs recommendations for settings common to most connection pooling software. If your application has multiple workers, each worker should be configured based on its own connections and workload.
 
-Creating the appropriate size pool of connections is critical to gaining maximum performance in an application. Too few connections in the pool will result in high latency as each operation waits for a connection to open up. But adding too many connections to the pool can also result in high latency as each connection thread is being run in parallel by the system. The time it takes for many threads to complete in parallel is typically higher than the time it takes a smaller number of threads to run sequentially.
+### Set the maximum lifetime of connections
+
+The maximum lifetime of a connection should be set to no less than 5 minutes and no more than 30 minutes. Applications that connect to {{ site.data.products.serverless }} clusters should set this to 30 minutes, while applications that connect to {{ site.data.products.dedicated }} and {{ site.data.products.core }} clusters should set this to 5 minutes.
```

Comment:
Ideally, no. But the fact is that 30 minutes is the right setting for Serverless, given there are no load balancing issues or rolling restart issues. The only reason 5 minutes is being suggested for Dedicated/self-hosted is because of those products' limitations. 5 minutes means a lot of unnecessary cycling of connections, with higher risk of elevated tail latency and higher TLS CPU costs for the customer.

Reply:
> Do we want different recommendations between self-hosted, Dedicated, and Serverless?

Maybe in the future. For now, I say let's keep it simple and explain the tradeoffs like Rafi suggested. We can document what our server side timeouts are (soon to be changed for Dedicated) and build from there. I would suggest:

> In CockroachDB Dedicated, the nodes are configured to wait as long as 31 minutes for clients to terminate their connections before initiating a restart or shutdown. We recommend setting your client's maximum connection lifetime to 30 minutes or less, and your choice should be guided by your application's specific needs:
> 1. **Choose a Longer Lifetime for Stability and Lower Latency:** If your application benefits from stable connections and reduced frequency of connection attempts, setting a longer maximum connection lifetime can be beneficial. This can improve connection experience for CockroachDB Dedicated cluster updates by minimizing disruptions caused by node restarts.
> 2. **Opt for a Shorter Lifetime for Efficient Load Balancing:** If your application requires the ability to quickly redistribute load across newly introduced nodes in the cluster, a shorter maximum connection lifetime is advisable. Note that in CockroachDB Serverless, this is less of a concern as the system is designed to automatically balance client connections.

Reply:
For now, let's keep this as is. And the change I am suggesting above is better done as part of [CC-25475](https://cockroachlabs.atlassian.net/browse/CC-25475). I'll leave a comment there.